ql/src/java/org/apache/hadoop/hive/ql/parse/TypeCheckProcFactory.java

```diff
@@ -1220,16 +1220,26 @@ protected ExprNodeDesc getXpathOrFuncExprNodeDesc(ASTNode expr,
             }
             outputOpList.add(nullConst);
           }
+
           if (!ctx.isCBOExecuted()) {
-            ArrayList<ExprNodeDesc> orOperands = TypeCheckProcFactoryUtils.rewriteInToOR(children);
-            if (orOperands != null) {
-              if (orOperands.size() == 1) {
-                orOperands.add(new ExprNodeConstantDesc(TypeInfoFactory.booleanTypeInfo, false));
+
+            HiveConf conf;
+            try {
+              conf = Hive.get().getConf();
```

Comment:
Instead of obtaining the conf object here statically, let's just pass the int value using the ctx.

Reply:
@jcamachor `ctx` is passed on to `genExprNode` by the callers. There are several callers which uses `genExprNode`. For this we will have to update all of these callers to set the config in `TypeCheckCtx`. 
Fetching HiveConf only occurs if the expression at hand is  `GenericUDFIn` and CBO wasn't executed therefore this shouldn't add much overhead.
Let me know if you think otherwise

Reply:
I think it is better to pass this value from the callers in the context. You would not need to change all callers; if value is not passed, rewriting could be skipped. I see mainly two advantages of doing this:
1) if transformation is never happening, we will not be retrieving the conf and this value for every IN clause in a query (note that `isCBOExecuted` method is misleading, the value returned is `foldExpr` boolean which is `false` sometimes even for calls coming from CBO cf. first line in `genFilterRelNode` method in `CalcitePlanner`), and
2) removing the static call to Hive object from within the folding logic.
I see there are other calls to `Hive.get()` in the class, that information should probably be moved to context too.
These can all be tackled together in a follow-up, but I think since we are cleaning up this logic, it would make sense to do it at some point.

Reply:
Ok. I'll create a separate jira and tackle this in a follow-up.

Reply:
@jcamachor I have created https://issues.apache.org/jira/browse/HIVE-22135 for refactoring Hive.get().getConf().
For the current patch I have gotten green run. Let me know if everything looks good to you.